The file stockton4.dat contains data on 1,500 houses sold in Stockton, CA during 1996–1998. Variable descriptions are in the file stockton4.def.

1. Estimate the following model
$$ \ln(SPRICE) = \beta_1 +\beta_2 LIVAREA + \beta_3 LIVERA^2 + \beta_4 AGE + \beta_5 AGE^2$$
$$ + \beta_6 BEDS + \beta_7 (LIVAREA \times BEDS) + \beta_8 (LIVAREA^2\times BEDS)$$
$$ + \beta_9 (AGE \times BEDS)  + \beta_{10} (AGE^2 \times BEDS) + e$$

Report the estimated relatinoship between $\ln(SPRICE), LIVAREA$ and $AGE$ for two-, three- and four-bedroom houses.



In [15]:
clear all 

use https://www.stata.com/data/s4poe4/stockton4.dta

gen log_sprice = log(sprice)
gen livarea2 = livarea^2
gen age2 = age^2
gen livareabeds = livarea * beds
gen livarea2beds = livarea^2 * beds
gen agebeds = age*beds
gen age2beds = age^2 * beds

In [10]:
reg log_sprice livarea livarea2 age age2 beds livareabeds livarea2beds agebeds age2beds


      Source |       SS           df       MS      Number of obs   =     1,500
-------------+----------------------------------   F(9, 1490)      =    354.80
       Model |  147.934591         9  16.4371768   Prob > F        =    0.0000
    Residual |  69.0291994     1,490  .046328322   R-squared       =    0.6818
-------------+----------------------------------   Adj R-squared   =    0.6799
       Total |  216.963791     1,499   .14473902   Root MSE        =    .21524

------------------------------------------------------------------------------
  log_sprice |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
     livarea |      .1268   .0212512     5.97   0.000     .0851146    .1684854
    livarea2 |  -.0012677   .0005148    -2.46   0.014    -.0022775    -.000258
         age |   -.016916   .0073727    -2.29   0.022    -.0313779   -.0024541
        age2 |   .0002939    .000125     2.35   0.

In [12]:
di "Two bedrooms"
reg log_sprice livarea livarea2 age age2 beds livareabeds livarea2beds agebeds age2beds if beds == 2


Two bedrooms

note: livarea omitted because of collinearity
note: livarea2 omitted because of collinearity
note: age omitted because of collinearity
note: age2 omitted because of collinearity
note: beds omitted because of collinearity

      Source |       SS           df       MS      Number of obs   =        77
-------------+----------------------------------   F(4, 72)        =     20.67
       Model |   5.4841192         4   1.3710298   Prob > F        =    0.0000
    Residual |  4.77630645        72   .06633759   R-squared       =    0.5345
-------------+----------------------------------   Adj R-squared   =    0.5086
       Total |  10.2604257        76  .135005601   Root MSE        =    .25756

------------------------------------------------------------------------------
  log_sprice |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
     livarea |          0  (omitted)
    livarea2 |

In [15]:
di "Three bedrooms"
reg log_sprice livarea livarea2 age age2 beds livareabeds livarea2beds agebeds age2beds if beds == 3


Three bedrooms

note: livarea omitted because of collinearity
note: livarea2 omitted because of collinearity
note: age omitted because of collinearity
note: age2 omitted because of collinearity
note: beds omitted because of collinearity

      Source |       SS           df       MS      Number of obs   =       969
-------------+----------------------------------   F(4, 964)       =    329.97
       Model |  55.4338934         4  13.8584733   Prob > F        =    0.0000
    Residual |  40.4869482       964  .041998909   R-squared       =    0.5779
-------------+----------------------------------   Adj R-squared   =    0.5762
       Total |  95.9208415       968  .099091778   Root MSE        =    .20494

------------------------------------------------------------------------------
  log_sprice |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
     livarea |          0  (omitted)
    livarea2

In [14]:
di "Four bedrooms"
reg log_sprice livarea livarea2 age age2 beds livareabeds livarea2beds agebeds age2beds if beds == 4


Four bedrooms

note: livarea omitted because of collinearity
note: livarea2 omitted because of collinearity
note: age omitted because of collinearity
note: age2 omitted because of collinearity
note: beds omitted because of collinearity

      Source |       SS           df       MS      Number of obs   =       400
-------------+----------------------------------   F(4, 395)       =    241.50
       Model |  46.0684072         4  11.5171018   Prob > F        =    0.0000
    Residual |  18.8376846       395  .047690341   R-squared       =    0.7098
-------------+----------------------------------   Adj R-squared   =    0.7068
       Total |  64.9060919       399  .162671909   Root MSE        =    .21838

------------------------------------------------------------------------------
  log_sprice |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
     livarea |          0  (omitted)
    livarea2 

2. Test the null hypothesis $H_0:\beta_6 = 0, \beta_8 = 0, \beta_9 = 0, \beta_{10} = 0$. Use $\alpha = 0.05$

We have to use the F-statistic this time. From below we see that the restricted model has a $SSE_{R} = 69.2467085$, and from above we see the unrestricted model has $SSE_{U} = 69.0291994$. With 4 restricitions, $J = 4$. $N = 1500$, the number of observations and $K = 10$, the number of betas.

The $F$-statistic is 
$$ F = \frac{(SSE_R - SSE_U)/J}{SSE_U / (N-K)} = \:\frac{\left(69.2467085\:-69.0291994\right)/4\:}{69.0291994\:/\:\left(1500-10\right)} = 1.17373 $$

And the rejection region for a significance level of $\alpha = 0.05$ is (from an online calculator)

$$ F_{(0.95, 4, 1490)} = 2.37790047$$

Since $F <F_{0.95} $, we fail to reject the null hypothesis.  

3. Estimate the model implied by the test reslt in (b).  Report the estimated relationship
 between $\ln(SPRICE), LIVAREA$ and $AGE$ for two-, three- and four-bedroom houses.
 
Since the results state that those betas are not statistically significant, we can include only those betas that were found to be significant, so $\{\beta_{i}  | i = 1, 2, 3, 4, 5, 7 \}$ and now we have the restricted model $$ \ln(SPRICE) = \beta_1 +\beta_2 LIVAREA + \beta_3 LIVERA^2 + \beta_4 AGE + \beta_5 AGE^2 + \beta_7 (LIVAREA \times BEDS) $$

In [5]:
reg log_sprice livarea livarea2 age age2 livareabeds


      Source |       SS           df       MS      Number of obs   =     1,500
-------------+----------------------------------   F(5, 1494)      =    637.40
       Model |  147.717082         5  29.5434165   Prob > F        =    0.0000
    Residual |  69.2467085     1,494  .046349872   R-squared       =    0.6808
-------------+----------------------------------   Adj R-squared   =    0.6798
       Total |  216.963791     1,499   .14473902   Root MSE        =    .21529

------------------------------------------------------------------------------
  log_sprice |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
     livarea |   .0901161   .0049032    18.38   0.000     .0804982     .099734
    livarea2 |  -.0003482   .0000943    -3.69   0.000    -.0005331   -.0001633
         age |  -.0080479   .0011784    -6.83   0.000    -.0103595   -.0057364
        age2 |   .0001424     .00002     7.13   0.

In [6]:
reg log_sprice livarea livarea2 age age2 beds livareabeds livarea2beds agebeds age2beds


      Source |       SS           df       MS      Number of obs   =     1,500
-------------+----------------------------------   F(9, 1490)      =    354.80
       Model |  147.934591         9  16.4371768   Prob > F        =    0.0000
    Residual |  69.0291994     1,490  .046328322   R-squared       =    0.6818
-------------+----------------------------------   Adj R-squared   =    0.6799
       Total |  216.963791     1,499   .14473902   Root MSE        =    .21524

------------------------------------------------------------------------------
  log_sprice |      Coef.   Std. Err.      t    P>|t|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
     livarea |      .1268   .0212512     5.97   0.000     .0851146    .1684854
    livarea2 |  -.0012677   .0005148    -2.46   0.014    -.0022775    -.000258
         age |   -.016916   .0073727    -2.29   0.022    -.0313779   -.0024541
        age2 |   .0002939    .000125     2.35   0.